service/pushbullet/pushbullet.go

```diff
@@ -0,0 +1,58 @@
+package pushbullet
+
+import (
+	"github.com/cschomburg/go-pushbullet"
+	"github.com/pkg/errors"
+)
+
+// Pushbullet struct holds necessary data to communicate with the Pushbullet API.
+type Pushbullet struct {
+	client          *pushbullet.Client
+	deviceNicknames []string
+}
+
+// New returns a new instance of a Pushbullet notification service.
+// For more information about Pushbullet api token:
+//    -> https://docs.pushbullet.com/#api-overview
+func New(apiToken string) *Pushbullet {
+	client := pushbullet.New(apiToken)
+
+	pb := &Pushbullet{
+		client:          client,
+		deviceNicknames: []string{},
+	}
+
+	return pb
+}
+
+// AddReceivers takes Pushbullet device nicknames and adds them to the internal deviceNicknames list.
+// The Send method will send a given message to all those devices.
+func (pb *Pushbullet) AddReceivers(deviceNicknames ...string) {
+
+	pb.deviceNicknames = append(pb.deviceNicknames, deviceNicknames...)
+}
+
+// Send takes a message subject and a message body and sends them to all valid devices.
+// you will need Pushbullet installed on the relevant devices
+// (android, chrome, firefox, windows)
+// see https://www.pushbullet.com/apps
+func (pb Pushbullet) Send(subject, message string) error {
+
+	for _, deviceNickname := range pb.deviceNicknames {
+
+		dev, err := pb.client.Device(deviceNickname)
+
+		if err != nil {
+			// Unregistered devices should not result in a failure for all others
+			continue
```

Comment:
I like your solution here, but I'm not a big fan of us completely ignoring this potential error. 

~Maybe we should unify all occurring errors into one 'big' error and return that at the end? Hm.. I don't want the send message to cancel when only one device fails but I also unlike ignoring errors.~

~We need a sleek solution for this that doesn't crash the function and reports errors at the same time.~

For now we're gonna return an error here instead of just skipping it. We're doing the same in other services too, I want to keep things even until we find a better solution. But I agree with you, we need a solution that collects/reports errors but is also more robust - robust meaning that other devices shouldn't be affected by a broken device.

Reply:
I'll update it to return an error